### PR TITLE
feat: support string value

### LINKED ISSUE #126: Color Prop not working as expected with false-y values
**Describe the bug** A clear and concise description of what the bug is.

The color prop does not activate with `value=0`. My guess is you are doing something like `"if value, apply color"` but it should probably be more like `"if value !== undefined, apply color"`.

As a feature request, I would love it if you could also update value to be either `number | string`.

**To Reproduce** Steps to reproduce the behavior:

pretty straightforward, give any country a value of 0 and the color will not display.

**Expected behavior** A clear and concise description of what you expected to happen.

pretty straightforward, give any country a value of 0 and the color should display.
Give any country a value of `undefined` and the color should not display.
If you want to get fancy, it probably makes sense to have `undefined | null` not display the color.

### PR DESCRIPTION
**Pull request by kennethbigler**
This feature adds support for `string` and the value `0` for the `value` prop.
Currently the code only checks if the value is false-y, so it now supports a 0 value.
Other consumers of this worldmap (myself being one of them), may want to display non-numeric information next to the country name. Personally I would like to display world flags, which cannot be done at the moment.

fix #126

### CODE REVIEW

**Comment by Josh-Cena** [comment]
For tests, just add an example under `website/src/components` and its corresponding page under `website/src/pages/examples`.

I'll help you fix the TS errors.

**Comment by kennethbigler** [comment]
> For tests, just add an example under `website/src/components` and its corresponding page under `website/src/pages/examples`.
> 
> I'll help you fix the TS errors.
> 
> This is to fix #126, correct?

That is correct! And ya do let me know what to do to fix those errors, I can definitely make an example :)

**Comment by Josh-Cena** [comment]
The error is unrelated. I would fix that first and get back to you. Thanks!

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
I've made some refactors to make the code lint & compile.

Please:

1. Explain what a string `value` represents, because it's not immediately obvious from your issue description.

If the explanation makes sense and we agree to accept this API extension, please:

2. Add docs.
3. Add an example. Especially highlighting how a value of `0` differs from a value of `undefined`.

**Comment by kennethbigler** [comment]
@Josh-Cena I have updated the description, let me know if that makes sense! I am happy to elaborate if needed.

With your approval I will continue on with steps 2 and 3 :)

**Comment by Josh-Cena** [comment]
I see. Couldn't you simply change the tooltip text in that case, though?

https://yanivam.github.io/react-svg-worldmap/examples/localization

**Comment by Josh-Cena** [comment]
Yes, I agree that `value` should be easier to use than `tooptipTextFunction`. Feel free to add docs and an example! (And do mention that using a string `value` is just a simpler way of using `tooptipTextFunction`)

**Comment by Josh-Cena** [comment]
I'm thinking about several things:

1. We should make `WorldMap` a generic function, so that we can avoid typing the callback as always passing `value: number | string` when in practice most users would not have string values.
2. When we have a mix of both, the strings should be treated as having value `0`.
3. We should have some kind of normalization algorithm for the colors. I still think having `0` mean "no color" is a valuable thing to do, although it falls apart if you have negative values. Maybe we need a logarithmic color scale.

**Comment by kennethbigler** [comment]
> I'm thinking about several things:
> 
> 1. We should make `WorldMap` a generic function, so that we can avoid typing the callback as always passing `value: number | string` when in practice most users would not have string values.
> 2. When we have a mix of both, the strings should be treated as having value `0`.
> 3. We should have some kind of normalization algorithm for the colors. I still think having `0` mean "no color" is a valuable thing to do, although it falls apart if you have negative values. Maybe we need a logarithmic color scale.

1. I am not sure where you mean for this, can you elaborate?
2. I am definitely happy to make `string` be treated as a `0` value.
3. especially as you say to support negative values, I really don't think it is that hard for users to set value to `undefined` to get no color.

**Comment by Josh-Cena** [comment]
The error is really messed up. @yanivam did not push the 2.0.0-alpha.1 publish commit to master, so the website is not picking up the local version. I would manually push a commit.

**Comment by kennethbigler** [comment]
@Josh-Cena it looks like the error is from the fact that website is looking for an npm build. I am going to try updating to local linking instead.

As for the point #2, should I make string be 0, or `minValue`? As you said, if we support negative values that could be tricky. I am thinking `minValue` would be better.

**Comment by Josh-Cena** [comment]
> it looks like the error is from the fact that website is looking for an npm build

The problem is because the website is depending on a newer version of react-svg-worldmap than what the workspace has (because @yanivam didn't push the commit where he published 2.0.0-alpha.1). I've fixed it in https://github.com/yanivam/react-svg-worldmap/commit/6493953d32407d2f5e0f0022d6a39d75b5b2cc32. Now it surfaces another bug in my lint workflow.

**Comment by Josh-Cena** [comment]
> should I make string be 0, or `minValue`? As you said, if we support negative values that could be tricky. I am thinking `minValue` would be better.

Yeah, I'm not sure. I would really have to think about the semantics. For one thing: `minValue` would still have a color. `0` was originally designed to have absolutely no color. That's why I said we probably need a normalization algorithm.

**Comment by Josh-Cena** [comment]
OK, making it a generic function works pretty nicely. If you directly pass a literal array, you get the additional benefit of inferring literal types instead of plain `string | number`.

**Comment by Josh-Cena** [comment]
@kennethbigler No need to link locally... I've already fixed this. I'll revert your last commit because it overwrote mine.

**Comment by kennethbigler** [comment]
> @kennethbigler It's looking good from me. I don't have any more reviews. I'll think about the semantics of having "0" mean "no color" in the coming days, and probably merge this next week. Feel free to share your thoughts as well.

@Josh-Cena I gave my opinion above, I think it would be best that if you want no color, you should change the value to `undefined`.

**Comment by yanivam** [comment]
Whoa, saw this thread and catching up now. @Josh-Cena and @kennethbigler do you all need me to do something? I can publish a new version if need be. Let me know what needs to be done on my end. This update looks good to me.

**Comment by kennethbigler** [comment]
Thanks @yanivam! I don't think so. 
Just excited for this to be available and waiting on @Josh-Cena to decide the exact alg he is hoping for.
Big shout out to @Josh-Cena for all the awesome help 🎉

**Comment by kennethbigler** [comment]
@Josh-Cena any update?

File: docs/customization.md [hunk 1]
```diff
@@ -22,9 +22,12 @@ const stylingFunction = ({
   country,
   color,
 }: CountryContext) => {
-  const opacityLevel = countryValue
-    ? 0.1 + (1.5 * (countryValue - minValue)) / (maxValue - minValue)
-    : 0;
+  const calculatedValue =
+    typeof countryValue === "string" ? minValue : countryValue;
+  const opacityLevel =
+    countryValue !== undefined
```
Comment (Josh-Cena) [inline docs/customization.md:28]:
```suggestion
    calculatedValue !== undefined
```

File: docs/customization.md [hunk 2]
```diff
@@ -22,9 +22,12 @@ const stylingFunction = ({
...
   color,
 }: CountryContext) => {
-  const opacityLevel = countryValue
-    ? 0.1 + (1.5 * (countryValue - minValue)) / (maxValue - minValue)
-    : 0;
+  const calculatedValue =
+    typeof countryValue === "string" ? minValue : countryValue;
+  const opacityLevel =
+    calculatedValue !== undefined
+      ? 0.1 + (1.5 * (countryValue - minValue)) / (maxValue - minValue)
```
Comment (Josh-Cena) [inline docs/customization.md:29]:
```suggestion
      ? 0.1 + (1.5 * (calculatedValue - minValue)) / (maxValue - minValue)
```

**Pull request merged**